sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsConf.scala

```diff
@@ -930,6 +930,45 @@ object RapidsConf {
       |  --conf spark.executorEnv.LIBCUDF_KERNEL_CACHE_PATH="/tmp/cudf-$USER"
       |  ```
       |""".stripMargin)
+
+      printSectionHeader("RAPIDS-Plugin Cache Serializer")
+      // scalastyle:off line.size.limit
+      println("""
+      |  Apache Spark provides an important feature to cache intermediate data and provide
+      |  significant performance improvement while running multiple queries on the same data. There
+      |  are two ways to cache a Dataframe or a DataSet i.e. call `persist(storageLevel)` or
+      |  `cache()`. Calling `cache()` is the same as calling `persist(MEMORY_AND_DISK)`. There are
+      |  many articles online that can be read about caching and its benefits as well as when to
+      |  cache but as a rule of thumb we should identify the Dataframe that is being reused in a
+      |  Spark Application and cache it. Even if the system memory isn't big enough, Spark will
+      |  utilize disk space to spill over. To read more about what storage levels are available look
+      |  at `StorageLevel.scala` in Spark.
+      |
+      |  Starting in Spark 3.1.0 users can add their own cache serializer, if they desire, by
+      |  setting the `spark.sql.cache.serializer` configuration. This is a static configuration
+      |  that is set once for the duration of a Spark application which means that you can only set the conf
+      |  before starting a Spark application and cannot be changed for that application's Spark
+      |  session.
+      |
+      |  RAPIDS Accelerator for Apache Spark version 0.4+ has a serializer `com.nvidia.spark.rapids.shims.spark311.ParquetCachedBatchSerializer`
+      |  that is optimized to run on the GPU and uses parquet to compress data before caching it.
+      |  ParquetCachedBatchSerializer can be used independent of what the value of
+      |  `spark.rapids.sql.enabled` is. If it is set to true then the parquet compression will run
+      |  on GPU if possible, and importantly
+      |  `spark.sql.inMemoryColumnarStorage.enableVectorizedReader` will not be honored as the GPU
+      |  data is always read in as columnar. If `spark.rapids.sql.enabled` is set to false
+      |  the cached objects will still be compressed on the CPU as a part of the caching process.
+      |  Also note that parquet doesn't support CalendarIntervalType or NullType out of the box, but
+      |  ParquetCachedBatchSerializer does by decomposing intervals to struct containing the
+      |  months, days and microseconds and NullType to Int column containing nulls.
+      |
+      |  To use this serializer please run Spark
+      |  ```
+      |  spark-shell --conf spark.sql.cache.serializer=com.nvidia.spark.rapids.shims.spark311.ParquetCachedBatchSerializer"
```

Comment:
Will this documentation need to be updated for each version of Spark going forward?  

com.nvidia.spark.rapids.shims.spark311.ParquetCachedBatchSerializer

Reply:
hmm... I will change it so we don't have to keep updating it

Reply:
It is about compatibility and we are potentially going to have to keep updating it.

We eventually might need some kind of a chart to indicate the version of the serializer that is supposed to be used with a given version of Spark. But in the short term we can explain the version requirements.

This is very similar to the ShuffleManager implementations that we have.  The reason for this is because the API that a ShuffleManager or a CachedBatchSerializer has to implement is not stable (marked as a DeveloperApi or even worse in the case of ShuffleManager it is private to spark).

https://github.com/apache/spark/blob/5718d64f3104f7a24a9d4b619748bcca03031c48/sql/core/src/main/scala/org/apache/spark/sql/columnar/CachedBatchSerializer.scala#L46-L47

https://github.com/apache/spark/blob/5718d64f3104f7a24a9d4b619748bcca03031c48/core/src/main/scala/org/apache/spark/shuffle/ShuffleManager.scala#L34